validator/staker.go

```diff
@@ -161,24 +163,74 @@ func NewStaker(
 		withdrawDestination = common.HexToAddress(config.WithdrawDestination)
 	}
 	return &Staker{
-		L1Validator:         val,
-		l1Reader:            l1Reader,
-		strategy:            strategy,
-		baseCallOpts:        callOpts,
-		config:              config,
-		highGasBlocksBuffer: big.NewInt(config.L1PostingStrategy.HighGasDelayBlocks),
-		lastActCalledBlock:  nil,
-		withdrawDestination: withdrawDestination,
-		inboxReader:         inboxReader,
-		nitroMachineLoader:  nitroMachineLoader,
-		updatingModuleRoot:  false,
+		L1Validator:          val,
+		l1Reader:             l1Reader,
+		strategy:             strategy,
+		baseCallOpts:         callOpts,
+		config:               config,
+		highGasBlocksBuffer:  big.NewInt(config.L1PostingStrategy.HighGasDelayBlocks),
+		lastActCalledBlock:   nil,
+		withdrawDestination:  withdrawDestination,
+		inboxReader:          inboxReader,
+		nitroMachineLoader:   nitroMachineLoader,
+		updatingModuleRoot:   false,
+		firstModuleRootSeen:  common.Hash{},
+		moduleRootConsistent: true,
 	}, nil
 }
 
+func (s *Staker) Initialize(ctx context.Context) error {
+	err := s.L1Validator.Initialize(ctx)
+	if err != nil {
+		return err
+	}
+	return s.updateLatestWasmRoot(ctx)
+}
+
+func (s *Staker) updateLatestWasmRoot(ctx context.Context) error {
+	if s.blockValidator == nil {
+		return nil
+	}
+	if !s.updatingModuleRoot {
+		haveModuleRoot := s.blockValidator.GetCurrentWasmModuleRoot()
+		if haveModuleRoot == (common.Hash{}) {
+			s.updatingModuleRoot = true
```

Comment:
this is the only case where it's set to true? so if validator had a moduleroot initially it will never happen?